flutter/shell/platform/tizen/channels/platform_channel.cc

```diff
@@ -95,58 +99,68 @@ void PlatformChannel::HandleMethodCall(
     std::unique_ptr<MethodResult<rapidjson::Document>> result) {
   const std::string& method = method_call.method_name();
   const rapidjson::Document* arguments = method_call.arguments();
+  result_ = std::move(result);
 
   if (method == kSystemNavigatorPopMethod) {
     SystemNavigatorPop();
-    result->Success();
+    result_->Success();
   } else if (method == kPlaySoundMethod) {
     PlaySystemSound(arguments[0].GetString());
-    result->Success();
+    result_->Success();
   } else if (method == kHapticFeedbackVibrateMethod) {
     std::string type;
     if (arguments->IsString()) {
       type = arguments[0].GetString();
     }
     HapticFeedbackVibrate(type);
-    result->Success();
+    result_->Success();
   } else if (method == kGetClipboardDataMethod) {
     // https://api.flutter.dev/flutter/services/Clipboard/kTextPlain-constant.html
     // The API only supports the plain text format.
     if (strcmp(arguments[0].GetString(), kTextPlainFormat) != 0) {
-      result->Error(kUnknownClipboardFormatError,
-                    "Clipboard API only supports text.");
+      result_->Error(kUnknownClipboardFormatError,
+                     "Clipboard API only supports text.");
       return;
     }
-    GetClipboardData([result = result.release()](const std::string& data) {
-      rapidjson::Document document;
-      document.SetObject();
-      rapidjson::Document::AllocatorType& allocator = document.GetAllocator();
-      document.AddMember(rapidjson::Value(kTextKey, allocator),
-                         rapidjson::Value(data, allocator), allocator);
-      result->Success(document);
-      delete result;
-    });
+    if (!GetClipboardData([this](std::optional<std::string> data) {
+          if (!data.has_value()) {
+            result_->Error(kUnknownClipboardError,
+                           "Clipboard.GetData() failed.");
+            return;
+          }
+
+          rapidjson::Document document;
+          document.SetObject();
+          rapidjson::Document::AllocatorType& allocator =
+              document.GetAllocator();
+          document.AddMember(rapidjson::Value(kTextKey, allocator),
+                             rapidjson::Value(data.value(), allocator),
+                             allocator);
+          result_->Success(document);
+        })) {
+      result_->Error(kUnknownClipboardError, "Clipboard.GetData() failed.");
```

Comment:
No need to make `result` a member variable (although it's okay in this case because there's only one async method which is unlikely to be called twice simultaneously). Alternatively you can

```cpp
auto* result_ptr = result.release();
if (!GetClipboardData([result_ptr](...) {
  ...
  result_ptr->Success(document);
  delete result_ptr;
})) {
  result_ptr->Error(...);
  delete result_ptr;
}
```

Otherwise you can simply make `GetClipboardData` a void function and move the unique_ptr.

Reply:
i updated code. only the member variable was removed and result_ptr was passed.

Reply:
Missing `delete` at line 142.

Reply:
If I do not call Error() on line 142, the report on failure cases in GetClipboardData() will be missing.

Reply:
Sorry I update code